src/typegoose.ts

```diff
@@ -246,22 +247,42 @@ export function deleteModelWithClass<T, U extends AnyParamConstructor<T>>(cl: U)
  * ```
  */
 export function getDiscriminatorModelForClass<T, U extends AnyParamConstructor<T>>(
-  from: mongoose.Model<any>,
+  from: mongoose.Model<any> | mongoose.Schema.Types.DocumentArray,
   cl: U,
   id?: string
 ) {
   const name = getName(cl);
-  if (models.get(name)) {
-    return models.get(name) as ReturnModelType<U, T>;
-  }
+  // disabling this for sanity
+  // if (models.get(name)) {
+  //   return models.get(name) as ReturnModelType<U, T>;
+  // }
   const sch = buildSchema(cl) as mongoose.Schema & { paths: object };
 
   const discriminatorKey = sch.get('discriminatorKey');
   if (sch.path(discriminatorKey)) {
     sch.paths[discriminatorKey].options.$skipDiscriminatorCheck = true;
   }
 
-  const model = from.discriminator(name, sch, id ? id : name);
+  let model: mongoose.Model<mongoose.Document>;
+
+  id = id ? id : name;
+
+  // this check exists only for typescript type reasons
+  if (isModel(from)) {
+    model = from.discriminator(name, sch, id);
+  } else if (from instanceof mongoose.Schema.Types.DocumentArray) {
+    // This Implementation is a PROTOTYPE, it might change into an own function
+    model = (from as mongoose.Schema.Types.DocumentArray).discriminator(
+      name,
+      sch,
+      // @ts-ignore - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/39927
```

Comment:
waiting for this to resolve

Reply:
this is now merged, can you continue with this PR?

Reply:
@nsams this pr is made as another way to #98, and because i dont fully understand that feature, i wanted someone other to test/review it (@B-Stefan)